File: source/ts/src/core/masa.ts
import type { MasaInfo } from '../types/panchang';

/** Saura Masa: 0 = Mesha … 11 = Meena. */
export function computeMasa(
  siderealSunLon: number,
  nameResolver: (index: number) => string,
): MasaInfo {
  const index = Math.floor(siderealSunLon / 30);
  return {
    index,
    name: nameResolver(index),
  };
}

File: source/go/internal/core/masa.go
package core

import (
	"math"

	"github.com/ishankgupta95/panchang-ts/source/go/v5/internal/types"
)

func ComputeMasa(siderealSunLon float64, nameResolver func(index int) string) types.MasaInfo {
	index := int(math.Floor(siderealSunLon / 30))
	return types.MasaInfo{
		Index: index,
		Name:  nameResolver(index),
	}
}
